### PR TITLE
WIP: Queries on NDSparse

### PR DESCRIPTION
**Pull request by pranavtbhat**
SQL like queries on NDSparse arrays. I think a generalized (outer, inner, theta etc.) join can also be done, but it feels awkward using column indices instead of column names. 

Here's a demo:
```julia
 a = NDSparse([12,21,32], [52,41,34], [11,53,150])
# NDSparse{Int64,Tuple{Int64,Int64}}:
# (12,52) => 11
# (21,41) => 53
# (32,34) => 150

b = NDSparse([12,23,32], [52,43,34], [56,13,10])
# NDSparse{Int64,Tuple{Int64,Int64}}:
# (12,52) => 56
# (23,43) => 13
# (32,34) => 10

naturaljoin(a, b, +)
# NDSparse{Int64,Tuple{Int64,Int64}}:
# (12,52) => 67
# (32,34) => 160

select(a, 1=>x->x<30, 2=>x->x>40)
# NDSparse{Int64,Tuple{Int64,Int64}}:
# (12,52) => 11
# (21,41) => 53

filter(x->x>100, a)
# NDSparse{Int64,Tuple{Int64,Int64}}:
# (32,34) => 150
```

cc @shashi

### CODE REVIEW

**Comment by pranavtbhat** [comment]
Bump.

**Comment by ViralBShah** [comment]
I am going to merge for now.

**Comment by ViralBShah** [comment]
@shashi Please do review.

File: src/query.jl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+# Column-wise filtering (Accepts conditions as column-function pairs)
+# Example: select(arr, 1 => x->x>10, 3 => x->x!=10 ...)
+
+function Base.select(arr::NDSparse, conditions::Pair...)
+   indxs = 1:length(arr)
+   cols = arr.indexes.columns
+   for (c,f) in conditions
+      indxs = intersect(indxs, filter(i->f(cols[c][i]), indxs))
```
Comment (pranavtbhat) [inline src/query.jl]: The intersect is unnecessary. I'll remove this.

**Comment by shashi** [comment]
What @tanmaykm told was [Theta join](http://stackoverflow.com/questions/7870155/difference-between-a-theta-join-equijoin-and-natural-join) is the more general type of join. We should probably have that and then implement naturaljoin with it (or leave it as is if it's faster)

**Comment by tanmaykm** [comment]
I had later found `thetajoin` to be extremely slow and implemented `naturaljoin` separately.

**Comment by pranavtbhat** [comment]
Thetajoin will be awkward to implement. The user will have to specify columns on which the join is to be operated.

**Pull request merged**